### PR TITLE
Cleanup comments and docs in channel fee update logic

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Following up on @ariard's review at https://github.com/rust-bitcoin/rust-lightning/pull/985#pullrequestreview-730197117

### CODE REVIEW

**Review by ariard — COMMENTED** [review]
ACK dd3097ca-modulo-if-you-feel-it : https://github.com/rust-bitcoin/rust-lightning/pull/985#discussion_r689921787

**Comment by ariard** [comment]
ACK ce4e20d

**Comment by TheBlueMatt** [comment]
Reverted the additional `debug_assert` as it doesn't pass borrowck, which presumably is why it wasnt in the first pr either :)

**Comment by ariard** [comment]
re-ACK `dd3097c`

File: lightning/src/ln/channel.rs [hunk 1]
```diff
@@ -424,22 +424,18 @@ pub(super) struct Channel<Signer: Sign> {
...
+	// Because it follows the same commitment flow as HTLCs, `FeeUpdateState` is either `Outbound`
+	// or matches a subset of the `InboundHTLCOutput` variants. It is then updated/used when
+	// generating new commitment transactions with exactly the same criteria as inbound/outbound
+	// HTLCs with similar state.
 	pending_update_fee: Option<(u32, FeeUpdateState)>,
-	// update_fee() during ChannelState::AwaitingRemoteRevoke is hold in
-	// holdina_cell_update_fee then moved to pending_udpate_fee when revoke_and_ack
-	// is received. holding_cell_update_fee is updated when there are additional
-	// update_fee() during ChannelState::AwaitingRemoteRevoke.
+	// If an update_fee() call is made with ChannelState::AwaitingRemoteRevoke set, we place it
```
Comment (valentinewallace) [inline lightning/src/ln/channel.rs:434]: Believe that should be `send_update_fee()`, not `update_fee()` and below

**Review by valentinewallace — COMMENTED** [review]
Otherwise ACK

**Comment by TheBlueMatt** [comment]
Only change from ariard's ack is comment fixups suggested by val, will take after CI:

```
$ git diff-tree -U1 dd3097c 623da4da
diff --git a/lightning/src/ln/channel.rs b/lightning/src/ln/channel.rs
index b0f733454..f8cbc26a2 100644
--- a/lightning/src/ln/channel.rs
+++ b/lightning/src/ln/channel.rs
@@ -433,7 +433,7 @@ pub(super) struct Channel<Signer: Sign> {
 	pending_update_fee: Option<(u32, FeeUpdateState)>,
-	// If an update_fee() call is made with ChannelState::AwaitingRemoteRevoke set, we place it
-	// here instead of `pending_update_fee` in the same way as we place outbound HTLC updates in
+	// If a `send_update_fee()` call is made with ChannelState::AwaitingRemoteRevoke set, we place
+	// it here instead of `pending_update_fee` in the same way as we place outbound HTLC updates in
 	// `holding_cell_htlc_updates` instead of `pending_outbound_htlcs`. It is released into
 	// `pending_update_fee` with the same criteria as outbound HTLC updates but can be updated by
-	// further `update_fee` calls, dropping the previous holding cell update entirely.
+	// further `send_update_fee` calls, dropping the previous holding cell update entirely.
 	holding_cell_update_fee: Option<u32>,
$
```

**Pull request merged**